### PR TITLE
refactor: optimize homepage

### LINKED ISSUE #167: index route bugs
- [ ]  create a new event, refresh page, event not showing up in UI

- [ ]  on first load zero events show up in UI

Screenshot 2023-06-29 at 9 26 53 AM

### PR DESCRIPTION
**Pull request by iRohitGaur**
### What is the change?
- dependent on https://github.com/Real-Dev-Squad/calendar-API/pull/100
- we were making 3 sequential api calls to backend on homepage route which was significantly delaying the loading of the page as these calls were on server-side of remix
- optimized the login flow
- moved relevant details to cookies (username, calendarId)
- moved getEvents call from server-side to frontend

Provide a small description of what did you change and provide the reference to the issue ticket.

### Is it bug?

- Steps to repro
- Expected
- Actual

### \*Dev Tested?

- [ ] Yes
- [ ] No

#### Platforms

- [ ] Web

#### Browsers

- [ ] Chrome
- [ ] Safari
- [ ] Firefox

### Before / After Change Screenshots

> For visual or interaction changes. Can be video / screenshot.

Closes #167 , #182 , #187 , #195 , #197 , #198 , #202

### CODE REVIEW

File: app/routes/index.tsx [hunk 1]
```diff
@@ -8,97 +8,84 @@ import Calendar from '~/components/Calendar';
...
+    fetchEvents();
   }, []);
 
   return (
     <div className="">
       <div className="flex flex-col-reverse md:flex-row ">
         <Navbar />
         <div className="flex justify-center flex-grow">
-          <Calendar events={eventList.length ? eventList : parseEvents(events ?? [])} view={view} />
+          <Calendar events={events} view={view} />
```
Comment (harshith-venkatesh) [inline app/routes/index.tsx:88]: Are we handling scenarios when eventList is undefined?

Comment (iRohitGaur) [reply app/routes/index.tsx:88]: @harshith-venkatesh by default its an empty array and it wont be undefined as we only set it when we get it, else we toast the error.

Comment (harshith-venkatesh) [reply app/routes/index.tsx:88]: Ok great

File: app/constants/urls.constants.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ export const checkUsername = (HOST: string, username: string) =>
```
Comment (harshith-venkatesh) [inline app/constants/urls.constants.ts:3]: Are we using getUserSelfData and CalendarId call now?

**Review by harshith-venkatesh — APPROVED** [review]
All looks good @iRohitGaur , check for the requirements of the functions which are currently not being used

File: app/constants/urls.constants.ts (hunk 2 above)
Comment (iRohitGaur) [reply app/constants/urls.constants.ts:3]: @harshith-venkatesh they are not used as of now but we might need them in future so I haven't removed them.

**Pull request merged**